packager/signer/signer.go

```diff
@@ -464,46 +456,64 @@ func formatLinkHeader(preloads []*rpb.Metadata_Preload) (string, error) {
 	return strings.Join(values, ","), nil
 }
 
-// serveSignedExchange does the actual work of transforming, packaging and signed and writing to the response.
-func (this *Signer) serveSignedExchange(resp http.ResponseWriter, fetchResp *http.Response, signURL *url.URL, act string, transformVersion int64) {
-	// Cap with maxBodyLength to limit the memory that serveSignedExchange
-	// consumes per request. The serveSignedExchange implementation requires all
-	// the response body in memory, and doesn't work on streamed data.
-	fetchBody, err := ioutil.ReadAll(io.LimitReader(fetchResp.Body, maxBodyLength))
+type SXGParams struct {
+	signURL          *url.URL
+	act              string
+	transformVersion int64
+}
+
+// consumedFetchResp stores the fetch response in memory - including the
+// consumed body, not a stream reader. MICE requires the sender to process its
+// payload in reverse order
+// (https://tools.ietf.org/html/draft-thomson-http-mice-03#section-2.1). In an
```

Comment:
Mention the fact that `transformer.Process` requires the whole contents in memory, too.